test/conformance/ingress/util.go

```diff
@@ -844,12 +845,22 @@ func CreateDialContext(t *testing.T, ing *v1alpha1.Ingress, clients *test.Client
 }
 
 type RequestOption func(*http.Request)
+type ResponseExpectation func(response *http.Response) error
```

Comment:
you might want in theory response and error?
That way dial error will be covered.

Reply:
We had a different implementation with both response and error, but the expectation for response is separate from an error check. We didn't feel it should be grouped together.

Reply:
ook.